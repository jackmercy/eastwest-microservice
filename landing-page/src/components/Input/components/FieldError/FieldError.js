import React, { memo } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import useStyles from './fieldError.styles';

const FieldError = ({ className, error, ...restProps }) => {
  const classes = useStyles();

  return (
    <p className={classNames(classes.root, className)} {...restProps}>
      {error}
    </p>
  );
};

FieldError.propTypes = {
  error: PropTypes.node.isRequired,
  className: PropTypes.string,
};
FieldError.defaultProps = { className: '' };

export default memo(FieldError);
