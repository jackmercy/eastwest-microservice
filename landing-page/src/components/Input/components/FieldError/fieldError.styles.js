import { createUseStyles } from 'react-jss';

import { rem } from '../../../../utils/jss';
import COLORS from '../../../../constants/COLORS';
import BREAKPOINTS from '../../../../constants/BREAKPOINTS';

export default createUseStyles({
  root: {
    width: '100%',
    fontFamily: 'inherit',
    fontSize: rem(16),
    fontWeight: 400,
    lineHeight: rem(24),
    marginTop: rem(12),
    marginBottom: 0,
    color: COLORS.textRed,
  },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    root: { marginTop: rem(6), fontSize: rem(14), lineHeight: rem(20) },
  },
});
