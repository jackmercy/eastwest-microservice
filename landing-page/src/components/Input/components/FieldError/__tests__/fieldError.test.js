import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import FieldError from '../FieldError';

describe('Testing Input component > FieldError', () => {
  it('should render match snapshot', () => {
    const wrapper = shallow(<FieldError error="Some error" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
