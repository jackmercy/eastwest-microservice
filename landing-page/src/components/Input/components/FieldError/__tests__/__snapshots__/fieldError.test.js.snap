// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Input component > FieldError should render match snapshot 1`] = `
<p
  className="root-0-2-1"
>
  Some error
</p>
`;
